Analysis/Tasks/PWGCF/o2femtodream/femtoDreamProducerTask.cxx

```diff
@@ -0,0 +1,138 @@
+                                                   aod::pidTOFKa, aod::pidTOFPr, aod::pidTOFDe>>;
+} // namespace o2::aod
+
+struct femtoDreamProducerTask {
+
+  Produces<aod::FemtoDreamCollisions> outputCollision;
+  Produces<aod::FemtoDreamParticles> outputTracks;
+  Produces<aod::FemtoDreamDebugParticles> outputDebugTracks;
+
+  Configurable<bool> ConfDebugOutput{"ConfDebugOutput", true, "Debug output"};
+
+  /// Event cuts
+  FemtoDreamCollisionSelection colCuts;
+  Configurable<float> ConfEvtZvtx{"ConfEvtZvtx", 10.f, "Evt sel: Max. z-Vertex (cm)"};
+  Configurable<bool> ConfEvtTriggerCheck{"ConfEvtTriggerCheck", false, "Evt sel: check for trigger"};
+  Configurable<int> ConfEvtTriggerSel{"ConfEvtTriggerSel", kINT7, "Evt sel: trigger"};
+  Configurable<bool> ConfEvtOfflineCheck{"ConfEvtOfflineCheck", false, "Evt sel: check for offline selection"};
+
+  Filter colFilter = nabs(aod::collision::posZ) < ConfEvtZvtx;
+
+  FemtoDreamTrackSelection trackCuts;
+  Configurable<std::vector<float>> ConfTrkCharge{"ConfTrkCharge", std::vector<float>{-1, 1}, "Trk sel: Charge"};
+  Configurable<std::vector<float>> ConfTrkPtmin{"ConfTrkPtmin", std::vector<float>{0.4f, 0.6f, 0.5f, 0.2f}, "Trk sel: Min. pT (GeV/c)"};
+  Configurable<std::vector<float>> ConfTrkPtmax{"ConfTrkPtmax", std::vector<float>{4.05f, 999.f}, "Trk sel: Max. pT (GeV/c)"};
+  Configurable<std::vector<float>> ConfTrkEta{"ConfTrkEta", std::vector<float>{0.8f, 0.7f, 0.9f}, "Trk sel: Max. eta"};
+  Configurable<std::vector<float>> ConfTrkTPCnclsMin{"ConfTrkTPCnclsMin", std::vector<float>{80.f, 70.f, 90.f}, "Trk sel: Min. nCls TPC"};
+  Configurable<std::vector<float>> ConfTrkTPCfCls{"ConfTrkTPCfCls", std::vector<float>{0.7f, 0.83f, 0.9f}, "Trk sel: Min. ratio crossed rows/findable"};
+  Configurable<std::vector<float>> ConfTrkTPCsCls{"ConfTrkTPCsCls", std::vector<float>{0.1f, 160.f}, "Trk sel: Max. TPC shared cluster"};
+  Configurable<std::vector<float>> ConfTrkDCAxyMax{"ConfTrkDCAxyMax", std::vector<float>{0.1f, 999.f}, "Trk sel: Max. DCA_xy (cm)"}; /// here we need an open cut to do the DCA fits later on!
+  Configurable<std::vector<float>> ConfTrkDCAzMax{"ConfTrkDCAzMax", std::vector<float>{0.2f, 999.f}, "Trk sel: Max. DCA_z (cm)"};    /// open cut needed for v0s etc. also there we need a lower limit bit
+  /// \todo maybe we need to remove the PID from the general cut container and have a separate one, these are lots and lots of bits we need
+  Configurable<std::vector<float>> ConfTrkPIDnSigmaMax{"ConfTrkPIDnSigmaMax", std::vector<float>{3.5f, 3.f, 2.5f}, "Trk sel: Max. PID nSigma"};
+  Configurable<std::vector<int>> ConfTrkTPIDspecies{"ConfTrkTPIDspecies", std::vector<int>{o2::track::PID::Electron, o2::track::PID::Pion, o2::track::PID::Kaon, o2::track::PID::Proton, o2::track::PID::Deuteron}, "Trk sel: Particles species for PID"};
+
+  /// todo with configurables - how to make sure we use the smallest/largest value?
```

Comment:
you want the smallest and largest from the std::vector array above?
I think you can configure an additional configurable here and overwrite it in the `init` function. I.e. loop over the array and put the smallest.

Reply:
so something like this? I thought these filters were set up as one of the first things, I remember we had quite a lengthy discussion on this and for the same reason we also abandoned some of the rather nice features we had made up

Configurable<float> smallestSomething{ ... };
Filter filter = track::Something > smallestSomething;

init() {
  std::sort( .. vector .. );
  smallestSomething = vector.at(0);
..
}

Reply:
It does not need to be a configurable itself, it can simply be a data member that you sen in `init()` based on configurable vectors.

Reply:
Good point @aalkin , even better
Yes, @a-mathis , just the init() declaration is differently (you get also the InitContext, check corelations.cxx)

Reply:
@jgrosseo ah yes of course, this was just a sketch.   void init(InitContext&) will do, right?

Reply:
@aalkin @jgrosseo so now I tried running this, and it appears that only the values used to initialise that variable are used for the filter. So this appears not to be working, even though the minimum is properly found and set. Is there something conceptually wrong in my implementation?

Reply:
@a-mathis indeed, I missed the fact the you want to use it for the filter. For that it needs to be defined as a `Configurable`. The reasons is that filters are instantiated long before the configuration is available, and anything that is not a configurable will use the value it has at that point, while configurables will be updated later.

Reply:
Correction: right now `init()` is run after the filters are updated, so even with `Configurable` I will have to change that order to make your use case work. You can test this already by swapping two blocks in `AnalysisTask.h`, the one in lines 771-773 should be before the one in lines 755-769.

Reply:
@aalkin Thanks a lot for the clarification. Is this something that is planned to be done soon? Then I will for now increase the initialisation of the variable and once the AnalysisTask.h is changed we'll adopt accordingly, ok?

Reply:
@aalkin will do a PR for this very soon, I assume. So you can already consider it done for your PR. Do you still need to adopt something?

Reply:
@jgrosseo @aalkin but actually how do I overwrite the value of a configurable? This doesn't compile
Thanks!

Configurable<float> TrackMinSelPtMin{"TrackMinSelPtMin", 0.4f, "(automatic) Minimal pT selection for tracks"};
void init(InitContext&) {
      TrackMinSelPtMin = ... something   
}

Reply:
You also need to define it as a `MutableConfigurable`